### PR TITLE
Donor can upload a logo

### PR DESCRIPTION
**Pull request by swardbladpaulo**
[API: Donor can upload a logo](https://www.pivotaltracker.com/story/show/176626852)
### User Story
```
As an API
In order to let the donor display its brand
I want to let donor upload a logo
```
## Changes proposed in this pull request:

* Creating index in UserController
* Creating a serializer for UserUpdate
* Changing from slip_base64 to self.split_bas64
modify user_controller
## What  have we learned working on this feature:
* How to update profile information of a user
* Adding image
* Working more with the terminal
## Screenshots (Client)
Screenshot from 2021-02-01 12-00-54

### CODE REVIEW

File: app/controllers/api/user_controller.rb [hunk 1]
```diff
@@ -3,24 +3,31 @@ class Api::UserController < ApplicationController
...
 
   def show
-    render json: @user_profile
+    render json: @user_profile, serializer: UserUpdateSerializer
   end
 
   def update
     if user_profile_params.value?('')
-      render json: { message: 'Fields can not be empty' }, status: 400
+      render json: { message: ' Fields cannod be empty' }, status: 400
```
Comment (Ehugo2000) [inline app/controllers/api/user_controller.rb]: typo

File: app/controllers/api/user_controller.rb [hunk 2]
```diff
@@ -3,24 +3,31 @@ class Api::UserController < ApplicationController
...
     elsif @user_profile.persisted?
+      attach_image(@user_profile)
       @user_profile.update(user_profile_params)
-      render json: @user_profile
+      @user_profile.save
+      render json: @user_profile, serializer: UserUpdateSerializer
     else
       render json: { message: 'wrong wrong wrong' }
```
Comment (Ehugo2000) [inline app/controllers/api/user_controller.rb]: update error message

**Review by Ehugo2000 — COMMENTED** [review]
looking good :)

File: app/controllers/api/user_controller.rb (hunk 1 above)
Comment (swardbladpaulo) [reply app/controllers/api/user_controller.rb]: done

File: app/controllers/api/user_controller.rb (hunk 2 above)
Comment (swardbladpaulo) [reply app/controllers/api/user_controller.rb]: done

**Pull request merged**